crates/ruff_linter/src/rules/pycodestyle/rules/blank_lines.rs

```diff
@@ -397,13 +400,13 @@ struct LinePreprocessor<'a> {
 
 impl<'a> LinePreprocessor<'a> {
     fn new(
-        tokens: TokenKindIter<'a>,
+        tokens: &'a Tokens,
         locator: &'a Locator,
         indent_width: IndentWidth,
         cell_offsets: Option<&'a CellOffsets>,
     ) -> LinePreprocessor<'a> {
         LinePreprocessor {
-            tokens,
+            tokens: tokens.up_to_first_unknown().iter().peekable(),
```

Comment:
I remember that the implementation here is somewhat performance sensitive. I wonder if we should instead use the slice iterator directly and use an internal `peek` method that clones the slice and calls next on it to avoid the overhead of testing if `peeked` is set in every iteration.

Reply:
Yeah, I can look into it. I basically did that with the `TokenKindIter` :)